### PR TITLE
For #3384

### PR DESCRIPTION
**Pull request by ljyhust**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

For #3384

## Brief changelog

when update peer should update member bean

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java [hunk 1]
```diff
@@ -198,26 +199,35 @@ public void onEvent(InetUtils.IPChangeEvent event) {
...
+            if (!objEqual.equals(newMember, member)) {
                 newMember.setExtendVal(MemberMetaDataConstants.LAST_REFRESH_TIME, System.currentTimeMillis());
                 MemberUtils.copy(newMember, member);
                 // member data changes and all listeners need to be notified
                 NotifyCenter.publishEvent(MembersChangeEvent.builder().members(allMembers()).build());
             }
             return member;
         });
-        
+
```
Comment (KomachiSion) [inline core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java:230]: Please do not change the indent.

Please use nacos codestyle to reformat your change.

detail see source code `style/nacosCodeStyle.md.`

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by KomachiSion** [comment]
Please sign your CLA first.

File: core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java (hunk 1 above)
Comment (ljyhust) [reply core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java:237]: OK

File: common/src/main/java/com/alibaba/nacos/common/utils/ObjectEqual.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package com.alibaba.nacos.common.utils;
+
+/**
+ * Object equal check.
+ *
+ * @author lijinyang
+ */
+public interface ObjectEqual<T> {
```
Comment (chuntaojun) [inline common/src/main/java/com/alibaba/nacos/common/utils/ObjectEqual.java:24]: Object有一个equals的方法了

Comment (ljyhust) [reply common/src/main/java/com/alibaba/nacos/common/utils/ObjectEqual.java:24]: yes, it is. It reminds me that we can use Comparator interface replace it.

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftPeerSet.java [hunk 3]
```diff
@@ -112,6 +118,29 @@ public void remove(List<String> servers) {
      */
     public RaftPeer update(RaftPeer peer) {
         peers.put(peer.ip, peer);
+
```
Comment (chuntaojun) [inline naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftPeerSet.java:121]: 每个节点只能原地更新自己的元数据信息，其他节点的元数据信息更新已经在core模块处理了，这里不再需要

**Comment by chuntaojun** [comment]
这个就不需要这么复杂了，这里直接Map.equals是不是就可以了呢

**Review by chuntaojun — CHANGES_REQUESTED** [review]

**Comment by KomachiSion** [comment]
我也觉得这样实现过于复杂了，看下能否直接用equals方法来处理

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftPeerSet.java (hunk 3 above)
Comment (ljyhust) [reply naming/src/main/java/com/alibaba/nacos/naming/consistency/persistent/raft/RaftPeerSet.java:121]: OK, good idea for Map.equals
这块在后续迭代中有处理么，同时会更新member raft state信息？如果是这样，如你所说，这里不需要。
这个pull request 是不是可以撤了 ^-^

**Comment by KomachiSion** [comment]
@ljyhust Sorry, The issue has been fixed by #3982 .

**Comment by ljyhust** [comment]
ok, nice

**Pull request closed without merging**